pkg/devicerepository/grpc.go

```diff
@@ -60,13 +61,31 @@ func (dr *DeviceRepository) ensureBaseAssetURLs(models []*ttnpb.EndDeviceModel)
 
 const defaultLimit = 1000
 
-func requireApplication(ctx context.Context, ids *ttnpb.ApplicationIdentifiers) error {
-	return rights.RequireApplication(ctx, ids, ttnpb.Right_RIGHT_APPLICATION_DEVICES_READ)
+var errNotAuthenticated = errors.DefineUnauthenticated(
```

Comment:
The name of the error (`errNotAuthenticated`) should be very close to the ID (`no_rights_in_context`). You probably want just `errUnauthenticated`/`unauthenticated`.